src/main/app/case/CosApiClient.ts

```diff
@@ -107,6 +116,39 @@ export class CosApiClient {
     }
   }
 
+  public async updateCaseData(
+    user: UserDetails,
+    caseId: string,
+    partyDetails: Partial<PartyDetails>,
+    partyType: PartyType,
+    caseType: CaseType
+  ): Promise<CaseWithId> {
+    try {
+      const data = {
+        partyDetails,
+        partyType,
+        caseType,
+      };
+      const response = await Axios.post(
+        config.get('services.cos.url') + `/${caseId}/${EVENT_RESPONDENT_MIAM}/case-update`,
```

Comment:
On this event name, I dont think it should refer to MIAM. may be you have taken this up from existing code. it should be ideally the event for citizen allegation of harm. double check this one for me.